src/ui.rs

```diff
@@ -3048,40 +3048,60 @@ impl Component for SectionLineView<'_> {
     }
 
     fn draw(&self, viewport: &mut Viewport<Self::Id>, x: isize, y: isize) {
-        const NEWLINE_ICON: &str = "⏎";
-        let Self { line_key: _, inner } = self;
+        fn replacement_characters(character: char) -> Option<&'static str> {
+            match character {
+                // Characters end up writing over each-other and end up
+                // displaying incorrectly if ignored. Replacing tabs
+                // with a known length string fixes the issue for now.
+                '\t' => Some("→   "),
+                '\n' => Some("⏎"),
+                _ => None,
+            }
+        }
+
+        /// Split the line into a sequence of [`Span`]s where control characters are
+        /// replaced with styled [`Span`]'s and push them to the [`spans`] argument.
+        fn push_spans_from_line<'line>(line: &'line str, spans: &mut Vec<Span<'line>>) {
+            const CONTROL_CHARACTER_STYLE: Style = Style::new().fg(Color::DarkGray);
+
+            let mut last_index = 0;
+            // Find index of the start of each character to replace
+            for (idx, char) in line.match_indices(|char| replacement_characters(char).is_some()) {
+                // Push the string leading up to the character and the styled replacment string
+                if let Some(replacment_string) =
```

Comment:
```suggestion
                // Push the string leading up to the character and the styled replacement string
                if let Some(replacement_string) =
```

suggestion: fix typo